README.md

```diff
@@ -285,17 +261,45 @@ await expect(snapshotStream).toRerender()
 > [!TIP]
 >
 > If you don't want these matchers not to be automatically installed, you can
-> import from `@testing-library/react-render-stream` instead.
+> import from `@testing-library/react-render-stream/pure` instead.  
+> Keep in mind that if you use the `/pure` import, you have to call the
+> `cleanup` export manually after each test.
+
+## Usage side-by side with `@testing-library/react` or other tools that set `IS_REACT_ACT_ENVIRONMENT` or use `act`
+
+This library is written in a way if should not be used with `act`, and it will
+throw an error if `IS_REACT_ACT_ENVIRONMENT` is `true`.
 
-## A note on `act`.
+React Testing Library usually sets `IS_REACT_ACT_ENVIRONMENT` to `true`
+globally, and wraps some helpers like `userEvent.click` in `act` calls.
 
-You might want to avoid using this library with `act`, as `act`
-[can end up batching multiple renders](https://github.com/facebook/react/issues/30031#issuecomment-2183951296)
-into one in a way that would not happen in a production application.
+To use this library side-by-side with React Testing Library, we ship the
+`disableActEnvironment` helper to undo these changes temporarily.
 
-While that is convenient in a normal test suite, it defeats the purpose of this
-library.
+It returns a `Disposable` and can be used together with the `using` keyword to
+automatically clean up once the scope is left:
 
-Keep in mind that tools like `userEvent.click` use `act` internally. Many of
-those calls would only trigger one render anyways, so it can be okay to use
-them, but avoid this for longer-running actions inside of `act` calls.
+```ts
+test('my test', () => {
+  using _disabledAct = disableActEnvironment()
+
+  // your test code here
+
+  // as soon as this scope is left, the environment will be cleaned up
+})
+```
+
+If you cannot use `using`, you can also manually call the returned `cleanup`
+function:
```

Comment:
```suggestion
function. We recommend using `finally` to ensure the act environment is cleaned up if your test fails, otherwise it could leak between tests:
```

You might consider something like this addition somewhere in here so that others understand the importance of `finally` here, otherwise this might get ignored and cause other issues. Feel free to tweak my sentence and move it where it makes sense (perhaps a caution callout?), but I definitely think we should add a statement here explaining its importance.